### PR TITLE
feat: add access to the default search in replaceSearch on a collection

### PR DESCRIPTION
**Pull request by ghusse**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/datasource-toolkit/test/__factories__/schema/column-schema.ts [hunk 1]
```diff
@@ -21,6 +21,14 @@ export class ColumnSchemaFactory extends Factory<ColumnSchema> {
       filterOperators: new Set(MAP_ALLOWED_OPERATORS_FOR_COLUMN_TYPE.Uuid),
     });
   }
+
+  text(): ColumnSchemaFactory {
```
Comment (Thenkei) [inline packages/datasource-toolkit/test/__factories__/schema/column-schema.ts:25]: 👀

File: packages/datasource-customizer/src/decorators/search/collection.ts [hunk 2]
```diff
@@ -94,6 +99,15 @@ export default class SearchCollectionDecorator extends CollectionDecorator {
     return conditionTree;
   }
 
+  private generatePlainSearchFilter(
+    searchText: string,
+    options?: SearchOptions,
+  ): PlainConditionTree {
+    const conditionTree = this.generateSearchFilter(null, searchText, options);
```
Comment (Thenkei) [inline packages/datasource-customizer/src/decorators/search/collection.ts:106]: What is the first parameter ?

File: packages/datasource-customizer/src/decorators/search/collection-search-context.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ export default class CollectionSearchContext<
     public readonly generateSearchFilter: (
       searchText: string,
       options?: SearchOptions<S, N>,
-    ) => ConditionTree,
+    ) => TConditionTree<S, N>,
```
Comment (Thenkei) [inline packages/datasource-customizer/src/decorators/search/collection-search-context.ts:38]: `TConditionTree` is a `PlainConditionTree`?

**Review by Thenkei — COMMENTED** [review]
Let's test it

File: packages/datasource-customizer/src/decorators/search/collection.ts [hunk 4]
```diff
@@ -94,6 +99,15 @@ export default class SearchCollectionDecorator extends CollectionDecorator {
     return conditionTree;
   }
 
+  private generatePlainSearchFilter(
+    searchText: string,
+    options?: SearchOptions,
+  ): PlainConditionTree {
```
Comment (Thenkei) [inline packages/datasource-customizer/src/decorators/search/collection.ts:105]:
```suggestion
  private generatePlainSearchFilter(
    caller: Caller,
    searchText: string,
    options?: SearchOptions,
  ): PlainConditionTree {
```

File: packages/datasource-customizer/src/decorators/search/collection.ts (hunk 2 above)
Comment (Thenkei) [inline packages/datasource-customizer/src/decorators/search/collection.ts:106]:
```suggestion
    const conditionTree = this.generateSearchFilter(caller, searchText, options);
```

File: packages/datasource-customizer/src/decorators/search/collection.ts [hunk 5]
```diff
@@ -35,7 +36,11 @@ export default class SearchCollectionDecorator extends CollectionDecorator {
 
     // Implement search ourselves
     if (this.replacer || !this.childCollection.schema.searchable) {
-      const ctx = new CollectionSearchContext(this, caller, this.generateSearchFilter.bind(this));
+      const ctx = new CollectionSearchContext(
+        this,
+        caller,
+        this.generatePlainSearchFilter.bind(this),
```
Comment (Thenkei) [inline packages/datasource-customizer/src/decorators/search/collection.ts:42]:
```suggestion
        this.generatePlainSearchFilter.bind(this, caller),
```

**Review by Thenkei — APPROVED** [review]
- ✅&nbsp;Code review - Forget my concerns about `caller`. `toPlainObject` method doesn't need the `caller`.
  - ✅&nbsp; Use pure functions when possible
  - ✅&nbsp; Performance concerns
  - ✅&nbsp; Security concerns
  - ✅&nbsp; The PR tackle one subject only
- ✅&nbsp;Automatic tests
  - ✅&nbsp; Unit tests
  - ❓&nbsp; Integration tests
- ✅&nbsp;Manual tests
- ✅&nbsp;PR title
- ❔&nbsp;PR linked to the clickup task

File: packages/datasource-customizer/src/decorators/search/collection.ts (hunk 2 above)
Comment (ghusse) [reply packages/datasource-customizer/src/decorators/search/collection.ts]: You're right, thanks

File: packages/datasource-customizer/src/decorators/search/collection-search-context.ts (hunk 3 above)
Comment (ghusse) [reply packages/datasource-customizer/src/decorators/search/collection-search-context.ts:38]: `TConditionTree<S, N>`  is a `PlainConditionTree`  but with the declaration of authorized field names.

**Review by Thenkei — APPROVED** [review]
👌

File: packages/datasource-customizer/src/decorators/search/collection-search-context.ts (hunk 3 above)
Comment (Thenkei) [reply packages/datasource-customizer/src/decorators/search/collection-search-context.ts:38]: We should indeed add a rule for this. All contexts (user accissible interfaces) must use template type. 💪

**Pull request merged**